### DISCUSSION TITLE
Re-entrant release-do-nothing script 🙏🏻

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/iris#5328] [Ideas]
## 📰 Custom Issue

As a core dev and recent `iris` release manager, I totally :heart: the [release_do_nothing.py](https://github.com/SciTools/iris/blob/main/tools/release_do_nothing.py) script.

It gives me the confidence of getting the release right first time, so as such it's pretty priceless as it makes my job way easier.

However, I tend to not do the release in one sitting, end-to-end, so mid-release and coming back to it, I have to cycle through the `SECTIONS` of the script to get back to where I left off.

It would be handy to be able to jump in to the script, starting for a custom `SECTION` e.g., something like this:

```bash

> python release_do_nothing.py --section 10
```

Or extending the script so that it locally cached the stage of the release automatically so that you could hop back in seamlessly. If this was attractive as a workflow, I'd imaging that you'd also want a way of clearing/resettomg the cache e.g.,

```
> python release_do_nothing.py --clean
```

Thoughts?

**Comment by bjlittle** [comment]
What say ye, fellow v3.6.x release buddy @ESadek-MO ?

**Reply by ESadek-MO** [reply]
My memory is unreliable, but this might have been discussed offline, and I think it was decided that it was more effort than value?

In theory though, as a buddy I was using the do-nothing to follow you along and make sure I was checking for the right things, and I would've found this useful. Your first suggestion would certainly be the better in my mind, especially as a buddy. 

But yeah, resources and execution might be the issue here.

**Comment by trexfeathers** [comment]
It can certainly be frustrating!

We did make some concessions to this problem - the section numbers, and changing the `DONE` input to `Y` - so that it's quicker to return to your previous place.

I know it's not perfect, but I imagined that anything more sophisticated would take some more complex engineering, and I'm mindful of the 80/20 rule. To put it another way: this is time that we could spend on another Iris problem so we'd need to justify that this has greater importance.

**Reply by bjlittle** [reply]
> complex engineering

Really? 🤔

**Reply by rcomer** [reply]
https://xkcd.com/1205/

**Reply by trexfeathers** [reply]
> > complex engineering
> 
> Really? 🤔

If it's easy, and I haven't realised, I certainly wouldn't stand in anyone's way 👍

**Reply by trexfeathers** [reply]
It wasn't easy. I must have missed a trick 🙈

#6062

Anyway it gradually became clear that - easy or not - this was important to do. A script that is a long sequence, and encourages uses of `Ctrl+C`, was bound to be difficult to use. We also have other potential use cases outside Iris where the logic can be re-used.
